File: baselibrary/src/main/java/com/ju/baselibrary/widget/gloading/GlobalLoadingStatusView.java
package com.ju.baselibrary.widget.gloading;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.core.content.ContextCompat;

import com.ju.baselibrary.R;
import com.ju.baselibrary.callback.RetryClickListener;
import com.ju.baselibrary.utils.NetWorkUtil;


/**
 * simple loading status view for global usage
 *
 * @author billy.qi
 * @since 19/3/19 23:12
 */
@SuppressLint("ViewConstructor")
public class GlobalLoadingStatusView extends LinearLayout implements View.OnClickListener {
    private final RetryClickListener retryClickListener;
    private final TextView mTextView;
    // private final Runnable mRetryTask;
    private final ImageView mImageView;

    //    public GlobalLoadingStatusView(Context context, Runnable retryTask) {
//        super(context);
//        setOrientation(VERTICAL);
//        setGravity(Gravity.CENTER_HORIZONTAL);
//        LayoutInflater.from(context).inflate(R.layout.view_global_loading_status, this, true);
//        mImageView = findViewById(R.id.image);
//        mTextView = findViewById(R.id.text);
//        this.mRetryTask = retryTask;
//        setBackgroundColor(ContextCompat.getColor(context,R.color.white));
//    }
    public GlobalLoadingStatusView(Context context, RetryClickListener retryClickListener) {
        super(context);
        setOrientation(VERTICAL);
        setGravity(Gravity.CENTER_HORIZONTAL);
        LayoutInflater.from(context).inflate(R.layout.view_global_loading_status, this, true);
        mImageView = findViewById(R.id.image);
        mTextView = findViewById(R.id.text);
        this.retryClickListener = retryClickListener;
        setBackgroundColor(ContextCompat.getColor(context, R.color.white));
    }

    public void setMsgViewVisibility(boolean visible) {
        mTextView.setVisibility(visible ? VISIBLE : GONE);
    }

    public void setStatus(int status) {
        boolean show = true;
        View.OnClickListener onClickListener = null;
        int image = R.drawable.loading;
        int str = R.string.str_none;
        switch (status) {
            case Gloading.STATUS_LOAD_SUCCESS:
                show = false;
                break;
            case Gloading.STATUS_LOADING:
                str = R.string.loading;
                break;
            case Gloading.STATUS_LOAD_FAILED:
                str = R.string.load_failed;
                image = R.mipmap.icon_failed;
                Boolean networkConn = NetWorkUtil.isConnected(getContext());
                if (networkConn != null && !networkConn) {
                    str = R.string.load_failed_no_network;
                    image = R.mipmap.icon_no_wifi;
                }
                onClickListener = this;
                break;
            case Gloading.STATUS_EMPTY_DATA:
                str = R.string.empty;
                image = R.mipmap.icon_empty;
                break;
            default:
                break;
        }
        mImageView.setImageResource(image);
        setOnClickListener(onClickListener);
        mTextView.setText(str);
        setVisibility(show ? View.VISIBLE : View.GONE);
    }

    @Override
    public void onClick(View v) {
//        if (mRetryTask != null) {
//
//            mRetryTask.run();
//        }
        if (retryClickListener != null) {
            retryClickListener.retry();
        }
    }
}

File: baselib/src/main/java/com/hzy/baselib/widget/gloading/GlobalLoadingStatusView.kt
package com.hzy.baselib.widget.gloading

import android.annotation.SuppressLint
import android.content.Context
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView

import androidx.core.content.ContextCompat

import com.hzy.baselib.R
import com.hzy.baselib.listener.RetryClickListener
import com.hzy.baselib.util.NetWorkUtil


/**
 * simple loading status view for global usage
 *
 * @author billy.qi
 * @since 19/3/19 23:12
 */
@SuppressLint("ViewConstructor")
class GlobalLoadingStatusView
    (context: Context, private val retryClickListener: RetryClickListener?) : LinearLayout(context),
    View.OnClickListener {
    private val mTextView: TextView
    private val mImageView: ImageView

    init {
        orientation = LinearLayout.VERTICAL
        gravity = Gravity.CENTER_HORIZONTAL
        LayoutInflater.from(context).inflate(R.layout.view_global_loading_status, this, true)
        mImageView = findViewById(R.id.image)
        mTextView = findViewById(R.id.text)
        setBackgroundColor(ContextCompat.getColor(context, R.color.white))
    }

    fun setMsgViewVisibility(visible: Boolean) {
        mTextView.visibility = if (visible) View.VISIBLE else View.GONE
    }

    fun setStatus(status: Int) {
        var show = true
        var onClickListener: View.OnClickListener? = null
        var image = R.drawable.loading
        var str = R.string.str_none
        when (status) {
            Gloading.STATUS_LOAD_SUCCESS -> show = false
            Gloading.STATUS_LOADING -> str = R.string.loading
            Gloading.STATUS_LOAD_FAILED -> {
                str = R.string.load_failed
                image = R.mipmap.icon_failed
                val networkConn = NetWorkUtil.isConnected(context)
                if (networkConn != null && !networkConn) {
                    str = R.string.load_failed_no_network
                    image = R.mipmap.icon_no_wifi
                }
                onClickListener = this
            }
            Gloading.STATUS_EMPTY_DATA -> {
                str = R.string.empty
                image = R.mipmap.icon_empty
            }
            else -> {
            }
        }
        mImageView.setImageResource(image)
        setOnClickListener(onClickListener)
        mTextView.setText(str)
        visibility = if (show) View.VISIBLE else View.GONE
    }

    override fun onClick(v: View) {
        retryClickListener?.retry()
    }
}
